pkg/kubectl/cmd/util/helpers.go

```diff
@@ -461,3 +463,39 @@ func UpdateObject(info *resource.Info, updateFn func(runtime.Object) error) (run
 
 	return info.Object, nil
 }
+
+// AddCmdRecordFlag adds --record flag to command
+func AddRecordFlag(cmd *cobra.Command) {
+	cmd.Flags().Bool("record", false, "Record current kubectl command in the resource annotation.")
+}
+
+func GetRecordFlag(cmd *cobra.Command) bool {
+	return GetFlagBool(cmd, "record")
```

Comment:
It won't panic since `GetFlagBool` handles it; it'll log the following error `err accessing flag record for command get: flag accessed but not defined: record`.